src/main/java/reposense/parser/ArgsParser.java

```diff
@@ -203,15 +209,54 @@ public static CliArguments parse(String[] args) throws HelpScreenException, Pars
         }
     }
 
+    /**
+     * Returns a since date that is one month before {@code cliUntilDate} (if present) or one month before report
+     *  generation date otherwise.
+     */
+    private static Date getSinceDate(Optional<Date> cliUntilDate) {
+        Calendar cal = Calendar.getInstance();
+        cliUntilDate.ifPresent(cal::setTime);
+        cal.set(Calendar.HOUR_OF_DAY, 0);
+        cal.set(Calendar.MINUTE, 0);
+        cal.set(Calendar.SECOND, 0);
+        cal.set(Calendar.MILLISECOND, 0);
+        cal.add(Calendar.MONTH, -1);
+        return cal.getTime();
+    }
+
+    /**
+     * Returns date of report generation with time set to 23:59:59.
+     */
+    private static Date getUntilDate() {
+        Calendar cal = Calendar.getInstance();
+        cal.set(Calendar.HOUR_OF_DAY, 23);
+        cal.set(Calendar.MINUTE, 59);
+        cal.set(Calendar.SECOND, 59);
+        cal.set(Calendar.MILLISECOND, 0);
+        return cal.getTime();
+    }
+
     /**
      * Verifies that {@code sinceDate} is earlier than {@code untilDate}.
      *
      * @throws ParseException if {@code sinceDate} supplied is later than {@code untilDate}.
      */
-    private static void verifyDatesRangeIsCorrect(Optional<Date> sinceDate, Optional<Date> untilDate)
+    private static void verifyDatesRangeIsCorrect(Date sinceDate, Date untilDate)
             throws ParseException {
-        if (sinceDate.isPresent() && untilDate.isPresent() && sinceDate.get().getTime() > untilDate.get().getTime()) {
+        if (sinceDate.getTime() > untilDate.getTime()) {
             throw new ParseException(MESSAGE_SINCE_DATE_LATER_THAN_UNTIL_DATE);
         }
     }
+
+    /**
+     * Verifies that {@code sinceDate} is no later than the date of report generation.
+     *
+     * @throws  ParseException if {@code sinceDate} supplied is later than date of report generation.
+     */
+    private static void verifySinceDateIsValid(Date sinceDate) throws ParseException {
```

Comment:
I believe this can be verified in `SinceDateArgumentType` you created, so you don't need an extra method to do this.

Reply:
Thought of doing that, but I'm not sure if doing so will violate LSP

Reply:
I don't think LSP is something we must follow, we are already violating it in some sense (Like the `convert` in `SinceDateArgumentType` is overriding a non-abstract method). So, I still believe it is better to put it in `SinceDateArgumentType`. 
Your inputs? @eugenepeh @yong24s